server/controllers/contact_lists.js

```diff
@@ -1,24 +1,114 @@
 
 export function saveNewContactList(req, res, next) {
   const contactListParams = {
     name: req.body.name
   };
+  // const uploadedCsv = req.files.csv;
+  // console.log(uploadedCsv);
 
-  const uploadedCsv = req.files.csv;
+  function validateCSV() {
+    return {
+      attributes: {
+        contacts: [
+          {
+            first_name: 'Joe',
+            last_name: 'Shoe',
+            phone_number: '+11235678901',
+            email: 'joe@shoe.com',
+            external_id: 'test1234'
+          },
+          {
+            first_name: 'Sue',
+            last_name: 'Shoe',
+            phone_number: '+11235678901',
+            email: 'sally@shoe.com',
+            external_id: 'test1235'
+          },
+          {
+            first_name: 'Jeff',
+            last_name: 'Slipper',
+            phone_number: '+1235671234',
+            email: 'charlie@slipper.com',
+            external_id: 'test1236'
+          },
+          {
+            first_name: 'Julie',
+            last_name: 'Flipflop',
+            phone_number: '+1123561234',
+            email: 'frank@flipflop.com',
+            external_id: 'test1237'
+          }
+        ]
+      }
+    };
+  }
+
+  try {
+    // validateCSV(uploadedCsv);
+  } catch (error) {
+    console.log(`Error in validating CSV: ${error}`);
+  }
 
-  console.log(uploadedCsv);
+  const testContactListParams = validateCSV();
 
   return contactListsService.saveNewContactList(contactListParams)
     .then((contactList) => {
       if (contactList) {
+        const { id } = contactList.attributes;
+        // FOR TESTING:
+        const { contacts } = testContactListParams.attributes;
+        contacts.forEach(contact =>
+          contactsService.getContactByPhoneNumberAndFirstName(contact)
+            .then((checkContact) => {
+              if (checkContact) {
+                const { id: contact_id } = checkContact.attributes;
+                const params = {
+                  id: contact_id,
+                  ...contact
+                };
+                contactsService.updateContactById(params)
+                  .then(() => {
+                    contactListsService.addContactToContactList({ contact_id, id })
+                      .then(() => {
+                        console.log('Successfully added contact to contact list after updating contact');
+                      })
+                      .catch((err) => {
+                        console.log(`Error in adding contact to contact list after updating contact: ${err}`);
+                      });
+                  })
+                  .catch((err) => {
+                    console.log(`Error in updating contact: ${err}`);
+                  });
+              } else {
+                contactsService.saveNewContact(contact)
+                  .then((newContact) => {
+                    const { id: contact_id } = newContact.attributes;
+                    contactListsService.addContactToContactList({ contact_id, id })
```

Comment:
Hey @perfectlynormalbeast! Running into some async issues here since the updateContact does not return the contact, if we combine the promises, we need to gather the contact_id in two different locations in order to make the association to the list. May need your help to determine what's possible in terms of DRYing this up.